### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by bsansone**
Adds actual working auth flow. Stores authenticated user token in secure storage. On app boot if token is found in secure storage, navigate to landing page, otherwise if there is no token or they have logged out, they will be navigated to login screen.

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -35,38 +31,50 @@ import Event from "./src/screens/Event";
...
+// { name: "EditUser", component: EditUser },
+// { name: "EditSchool", component: EditSchool },
+// { name: "EditEvent", component: EditEvent },
+// { name: "EditTournament", component: EditTournament },
+// { name: "CreateEvent", component: CreateEvent },
+// { name: "CreateTeam", component: CreateTeam },
+// { name: "CreateTournament", component: CreateTournament },
+// ];
+
+const unauthoriztedScreens = [
```
Comment (Bkohler93) [inline App.js]: Typo here: `unauthoriztedScreens` -> `unauthorizedScreens`

File: App.js [hunk 2]
```diff
@@ -84,14 +92,96 @@ function HomeScreen({ navigation }) {
...
+      },
+    }),
+    []
+  );
+
+  if (state.isLoading) {
+    return <Text>Loading...</Text>;
+  }
+
+  const screens = !state.userToken ? unauthoriztedScreens : authorizedScreens;
```
Comment (Bkohler93) [inline App.js]: `unauthoriztedScreens` typo here as well.

File: App.js [hunk 3]
```diff
@@ -84,14 +92,96 @@ function HomeScreen({ navigation }) {
...
+        userToken = await SecureStore.getItemAsync("userToken");
+      } catch (e) {
+        // Restoring token failed
+      }
+
+      // After restoring token, we may need to validate it in production apps
+
+      // This will switch to the App screen or Auth screen and this loading
+      // screen will be unmounted and thrown away.
+      dispatch({ type: "RESTORE_TOKEN", token: userToken });
```
Comment (Bkohler93) [inline App.js]: should this be placed inside the try portion of the try-catch block above? Wouldn't we only want `RESTORE_TOKEN` case to run if a `userToken` is fetched successfully?

File: src/providers/auth.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+  React.useEffect(() => {
+    setAuthStatus(AUTH_STATUS.AUTHENTICATING);
+
+    return onIdTokenChanged(auth, async (authUser) => {
+      if (process.env.ENV !== "production") {
+        console.log("[AUTH] Token changed!");
+      }
+
+      if (!Boolean(authUser)) {
```
Comment (Bkohler93) [inline src/providers/auth.js:74]: How is `!Boolean(authUser)` different than using `!authUser` here?

**Review by Bkohler93 — CHANGES_REQUESTED** [review]

File: src/providers/auth.js (hunk 4 above)
Comment (bsansone) [reply src/providers/auth.js:74]: Its not much different, just a little bit easier to read imo

File: App.js (hunk 1 above)
Comment (bsansone) [reply App.js]: Fixed

**Review by Bkohler93 — APPROVED** [review]

**Review by Bkohler93 — COMMENTED** [review]
The filename is `Login.js` instead of `LogIn.js`.. not sure if this will cause the linkage error that the `add-dynamic-event` branch gets.

image

**Pull request merged**